sahi/models/base.py

```diff
@@ -9,20 +10,22 @@
 from sahi.utils.import_utils import is_available
 from sahi.utils.torch import select_device as select_torch_device
 
+logger = logging.getLogger(__name__)
+
 
 class DetectionModel:
     def __init__(
         self,
         model_path: Optional[str] = None,
         model: Optional[Any] = None,
         config_path: Optional[str] = None,
-        device: Optional[str] = None,
+        device: str = "cpu",
```

Comment:
Instead of hardcoding it to CPU, we should automatically select the best device (if cuda is available set it as cuda) if user doesn't provide anything. [`select_device`](https://github.com/obss/sahi/blob/c9469f5baf57d614eb36f1c8a063e38ec1c62146/sahi/utils/torch.py#L54) should be updated for this I believe.